src/parser/HLSTree.cpp

```diff
@@ -173,442 +173,469 @@ bool adaptive::CHLSTree::Open(std::string_view url,
   return true;
 }
 
-PLAYLIST::PrepareRepStatus adaptive::CHLSTree::prepareRepresentation(PLAYLIST::CPeriod* period,
-                                                                     PLAYLIST::CAdaptationSet* adp,
-                                                                     PLAYLIST::CRepresentation* rep,
-                                                                     bool update)
+bool adaptive::CHLSTree::PrepareRepresentation(PLAYLIST::CPeriod* period,
```

Comment:
Here github show a big mess, there are no functional code changes on PrepareRepresentation, the only difference is that has been decoupled per functionality.
The old `prepareRepresentation` now has been splitted on more methods:
- PrepareRepresentation: Act more or less as before
- DownloadChildManifest: To download the child manifest only
- ParseChildManifest: Parse the child manifest data
- PrepareSegments: Update the segments (to adjust/set "current segment")

now ParseChildManifest could be reused in future to parse in right way the media playlist on CHLSTree::ParseManifest (see todo), not tested now, but a step forward to achieve it